### DISCUSSION TITLE
fHeroes AI surrenders combat even when surrender immediately causes their side to lose

### DISCUSSION THREAD

**Discussion by Granfalegion** [ihhub/fheroes2#9336] [General]
See an example of this occurring here: https://www.twitch.tv/khelbenscointv/clip/ExcitedMoralLEDPeteZaroll-FDmzCUTNKRGpps7M

The defending AI is the last hero in the last castle owned by their side.  If they lose the combat, they are eliminated from the game.  There's no benefit to the AI to choose to surrender rather than fight on in this situation.  The only upside I can see here is fueled entirely by spite and designed to deny the attacking hero any chance at the last hero's artifacts.

Perhaps that's intentional, but the behavior feels wonky and anticlimactic.

**Comment by Granfalegion** [comment]
I believe Khelben's running v1.14 in the above clip.

**Comment by oleg-derevenetz** [comment]
> designed to deny the attacking hero any chance at the last hero's artifacts.

Yes, that's the reason, this behavior is intentional. Even if the AI player gets eliminated, it is not obliged to strengthen the winner to fight the remaining players.

> Perhaps that's intentional, but the behavior feels wonky and anticlimactic.

You can't always get what you want (C) :)
